packages/forms-web-app/tests/unit/lib/client-side/javascript-requiring-consent.test.js

```diff
@@ -37,39 +46,202 @@ describe('lib/client-side/javascript-requiring-consent', () => {
+        server: {
+          googleTagManagerId: 'some-test-value',
+        },
+        featureFlag: {
+          foo: true, // no googletagmanager feature flag.
+        },
+      };
+
+      initialiseOptionalJavaScripts(document, config);
+
+      // eslint-disable-next-line no-console
+      expect(console.log).toHaveBeenCalledWith(
+        'Declined consent. Third party cookies are not enabled.'
+      );
+
+      expect(initialiseGoogleAnalytics).not.toHaveBeenCalled();
+      expect(googleTagManager.denyConsent).not.toHaveBeenCalled();
+      expect(googleTagManager.grantConsent).not.toHaveBeenCalled();
+    });
+
+    test('return early if `usage=false and googleTagManager featureFlag false`', () => {
+      jest.spyOn(console, 'log').mockImplementation();
+
+      readCookie.mockImplementation(() => JSON.stringify({ usage: false }));
+
+      const config = {
+        server: {
+          googleTagManagerId: 'some-test-value',
+        },
+        featureFlag: {
+          googleTagManager: false, // no googletagmanager feature flag.
+        },
+      };
+
+      initialiseOptionalJavaScripts(document, config);
+
+      // eslint-disable-next-line no-console
+      expect(console.log).toHaveBeenCalledWith(
+        'Declined consent. Third party cookies are not enabled.'
+      );
+
+      expect(initialiseGoogleAnalytics).not.toHaveBeenCalled();
+      expect(googleTagManager.denyConsent).not.toHaveBeenCalled();
+      expect(googleTagManager.grantConsent).not.toHaveBeenCalled();
+    });
+
+    test('return early if `usage=false and googleTagManagerId is undefined`', () => {
+      jest.spyOn(console, 'log').mockImplementation();
+
+      readCookie.mockImplementation(() => JSON.stringify({ usage: false }));
+
+      const config = {
+        server: {
+          foo: 'some-test-value',
+        },
+        featureFlag: {
+          googleTagManager: true, // no googletagmanager feature flag.
+        },
+      };
+
+      initialiseOptionalJavaScripts(document, config);
+
+      // eslint-disable-next-line no-console
+      expect(console.log).toHaveBeenCalledWith(
+        'Declined consent. Third party cookies are not enabled.'
+      );
 
       expect(initialiseGoogleAnalytics).not.toHaveBeenCalled();
+      expect(googleTagManager.denyConsent).not.toHaveBeenCalled();
+      expect(googleTagManager.grantConsent).not.toHaveBeenCalled();
     });
 
-    test('return early if `usage=false`', () => {
+    test('Deny Consent to Google Tag manager then return early if `usage=false and googleTagManagerId and googleTagManager feature flag is true`', () => {
       jest.spyOn(console, 'log').mockImplementation();
 
       readCookie.mockImplementation(() => JSON.stringify({ usage: false }));
 
-      initialiseOptionalJavaScripts();
+      const config = {
+        server: {
+          googleTagManagerId: 'some-test-value',
+        },
+        featureFlag: {
+          googleTagManager: true, // no googletagmanager feature flag.
```

Comment:
This comment looks like it needs the 'no' removed :)

Reply:
Have removed 'no'